### PR TITLE
Migrate tree.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/tree.dart [hunk 1]
```diff
@@ -146,10 +144,10 @@ class TreeViewItem<T extends TreeNode<T>> extends StatefulWidget {
 
   final T data;
 
-  final Widget Function(VoidCallback onPressed) buildDisplay;
+  final Widget Function(VoidCallback onPressed)? buildDisplay;
 
-  final void Function(T) onItemSelected;
-  final void Function(T) onItemExpanded;
+  final void Function(T)? onItemSelected;
+  final void Function(T)? onItemExpanded;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/tree.dart:150]: make all of these non-nullable and required.  Looks like the TreeViewItem class can also be private. Let's do that while we are here.

File: packages/devtools_app/lib/src/shared/tree.dart [hunk 2]
```diff
@@ -26,7 +24,7 @@ class TreeView<T extends TreeNode<T>> extends StatefulWidget {
     this.emptyTreeViewBuilder,
   });
 
-  final ValueListenable<List<T>> dataRootsListenable;
+  final ValueListenable<List<T>>? dataRootsListenable;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart:27]: done

File: packages/devtools_app/lib/src/shared/tree.dart [hunk 3]
```diff
@@ -36,25 +34,25 @@ class TreeView<T extends TreeNode<T>> extends StatefulWidget {
   /// Defaults to false.
   final bool shrinkWrap;
 
-  final Widget Function(T, VoidCallback) dataDisplayProvider;
+  final Widget Function(T, VoidCallback)? dataDisplayProvider;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart:37]: done

File: packages/devtools_app/lib/src/shared/tree.dart [hunk 4]
```diff
@@ -36,25 +34,25 @@ class TreeView<T extends TreeNode<T>> extends StatefulWidget {
...
 
   /// Invoked when a tree node is selected. If [onItemExpanded] is not
   /// provided, this method will also be called when the expand button is
   /// tapped.
-  final FutureOr<void> Function(T) onItemSelected;
+  final FutureOr<void> Function(T)? onItemSelected;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart:42]: done

File: packages/devtools_app/lib/src/shared/tree.dart [hunk 5]
```diff
@@ -200,24 +198,24 @@ class _TreeViewItemState<T extends TreeNode<T>> extends State<TreeViewItem<T>>
...
     setExpanded(widget.data.isExpanded);
   }
 }
 
 mixin TreeMixin<T extends TreeNode<T>> {
-  List<T> dataRoots;
+  List<T>? dataRoots;
 
-  List<T> items;
+  List<T>? items;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart:214]: Are you sure? They are not initialized in the mixin

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/tree.dart:214]: They should be initialized though by any class mixing in this mixin. I would think making them late would force subclasses to initialize them. Or at least would throw an error if they don't.

File: packages/devtools_app/lib/src/shared/tree.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart]: done

File: packages/devtools_app/lib/src/shared/tree.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart]: done

File: packages/devtools_app/lib/src/shared/tree.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/tree.dart]: buildDisplay can be non-nullable too.

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/tree.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**